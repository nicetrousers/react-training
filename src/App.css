.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}
.App-edit {
  float: right;
}
.App-form {
  width: 50%;
  margin: 0 auto;
}
.App-header {
  background-color: #222;
  padding: 20px;
  color: white;
  height: 270px;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

ul {
  list-style-position: inside;
}