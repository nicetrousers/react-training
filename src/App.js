import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Bio from './Bio.js';

export default class App extends Component {
  render() {
    let portrait = "https://ca.slack-edge.com/T02540DGA-U0JCA4317-g0169d2f003a-72";
    let bullets = [
      "Works on Expectnation, the conferences app",
      "Tries not to react too sharply"
    ];
    return (
      <div className="App">
        <Bio name="Marcel M. Cary" portrait={portrait} bullets={bullets} />
      </div>
    );
  }
}

// export default App;