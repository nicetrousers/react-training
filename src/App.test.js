import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('passes 3', () => {});

window.fetch = () => Promise.resolve({
	json: () => Promise.resolve({
		name: "The Bio Name",
		portrait: "portrait",
		bullets: [
			"bullet",
			],
	}),
});

it('renders bio name with stubbed fetch', (done) => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  setTimeout(() => {
		const name = "The Bio Name"
  	expect(div.outerHTML).toEqual(expect.stringContaining(name));
  	done();
  }, 0);
});

it('renders bio snapshot', (done) => {
	const app = renderer.create(<App />);
  setTimeout(() => {
		expect(app.toJSON()).toMatchSnapshot();
	  done();
  }, 0);
});