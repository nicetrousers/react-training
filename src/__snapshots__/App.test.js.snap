// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bio snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div>
      <header
        className="App-header"
        onClick={[Function]}
      >
        <div
          className="App-edit"
          onClick={[Function]}
        >
          <span
            className="src-Icon-icon"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\"><title>O&amp;apos;ReillyArtboard 20</title><path d=\\"M19.12 2.29a1 1 0 0 0-1.41 0l-2 2 4 4 2-2a1 1 0 0 0 0-1.41zM3.59 16.41A2 2 0 0 0 3 17.83V21h3.17a2 2 0 0 0 1.41-.59l10.71-10.7-4-4z\\"/></svg>",
              }
            }
          />
        </div>
        <div
          className="src-Avatar-wrapper src-Avatar-large"
        >
          <a
            className="src-Avatar-anchor"
            href={undefined}
            onClick={undefined}
            target={null}
          >
            <img
              alt="avatar"
              className="src-Avatar-avatar src-Avatar-large"
              src="portrait"
            />
          </a>
        </div>
        <h1
          className="App-title"
        >
          The Bio Name
        </h1>
      </header>
      <div
        className="App-intro"
      >
        <ul>
          <li>
            bullet
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
